File: sast-python2/case/accuracy/context_sensitive/argument_return_value_passing/argument_passing_value_009_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 准确度->上下文敏感分析->参数/返回值传递
# scene introduction = 参数值传递->可变参数
# level = 2
# bind_url = accuracy/context_sensitive/argument_return_value_passing/argument_passing_value_009_T
# evaluation information end
import os

def argument_passing_value_009_T(taint_src):
    def wrapper(*args):
        taint_sink(args[0])  # 可变参数首元素识别
    
    wrapper(taint_src, u"safe_value")

def taint_sink(o):
    os.system(o)

if __name__ == u"__main__":
    taint_src = u"taint_src_value"
    argument_passing_value_009_T(taint_src)

File: sast-go/cases/completeness/single_app_tracing/function_call/argument_passing/argument_passing_value_009_T/argument_passing_value_009_T.go

// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->函数和方法调用->参数传递
// scene introduction = this
// level = 2
// bind_url = completeness/single_app_tracing/function_call/argument_passing/argument_passing_value_009_T/argument_passing_value_009_T
// evaluation information end

package main
import "os/exec"

type Person struct {
	Name string
}

func argument_passing_value_009_T(__taint_src string) {
	person := Person{Name: __taint_src}

	person.UpdateNamePointer("safe")
	__taint_sink(person.GetNamePointer())
}

func (p Person) UpdateNamePointer(newName string) {
	p.Name = newName
}

func (p Person) GetNamePointer() string {
	return p.Name
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
	}

func main() {
    __taint_src := "taint_src_value"
    argument_passing_value_009_T(__taint_src)
}